### DISCUSSION TITLE
Pass Auth Props To Nav

### DISCUSSION THREAD

**Discussion by cdt-eth** [supabase/supabase#1991] [Questions] [answered]
Hey!

I followed the [React Quick Start guide](https://supabase.io/docs/guides/with-react) and now I'm trying to get the logged in username to show up in the Nav. How do I pass that username to another component? 

How can I pass the info `async function updateProfile({ username, website, avatar_url }) {` to my Nav component? Also, I'd need it to rerender when logged out. So I guess add some `loggedIn` state in `Auth` component and pass it to `Nav` as well.

**Comment by cdt-eth** [comment]
is Context my only option here?

**Comment by ftonato** [comment, marked as answer]
Hey @christiandavidturner, how are you doing?

It's not so easy to give you the "correct" answer because it depends on your project context, so the correct way to solve this will depend on how you implemented your solution (you are 100% free to do as you like).

Let's try to explore possible solutions to your problem together:

- You can have a **NAV** component and access the session information, after that you just need to print the information you want.
- Or else in the **NAV** component, you can receive the session information as a parameter.

```javascript
// src/Nav.js
import './nav.css'
import { useState, useEffect } from 'react'
import { supabase } from './supabaseClient'

export default function Nav() {
  const [session, setSession] = useState(null)

  useEffect(() => {
    setSession(supabase.auth.session())

    supabase.auth.onAuthStateChange((_event, session) => {
      setSession(session)
    })
  }, [])

  return (
    <div className="nav">
        <span> { session ? 🟢 : 🔴 } </span>
    </div>
  )
}
```

Another important reminder, you can consume two different values, to know if there is session data and what user data is through the application:

```javascript
// Inside a browser context, user() will return the user data, if there is a logged in user = https://supabase.io/docs/reference/javascript/auth-user 
const user = supabase.auth.user()

// Returns the session data, if there is an active session = https://supabase.io/docs/reference/javascript/auth-session
const session = supabase.auth.session()
```

About your problem of re-rendering some components when logging out, you can do them using different approaches:

- As a callback of the method `supabase.auth.signOut()` (https://supabase.io/docs/reference/javascript/auth-signout)
- Watching the `session` values `supabase.auth.session()`
- Or even, as a callback of the method `supabase.auth.onAuthStateChange` (https://supabase.io/docs/gotrue/client/onauthstatechange)

I hope I have helped you!

**Reply by cdt-eth** [reply]
this was helpful thank you! didn't need context.
